cmd/submit.go

```diff
@@ -15,20 +26,143 @@ var submitCmd = &cobra.Command{
 
 The CLI will do its best to figure out what to submit.
 
-If you call the command without any arguments, it will check
-if the current directory is an exercise, and if so, submit that.
+If you call the command without any arguments, it will
+submit the exercise contained in the current directory.
 
 If called with the path to a directory, it will submit it.
 
 If called with the name of an exercise, it will work out which
 track it is on and submit it. The command will ask for help
 figuring things out if necessary.
-	`,
+`,
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("submit called")
+		usrCfg, err := config.NewUserConfig()
+		BailOnError(err)
+
+		cliCfg, err := config.NewCLIConfig()
+		BailOnError(err)
+
+		if len(args) == 0 {
+			cwd, err := os.Getwd()
+			BailOnError(err)
+			args = []string{cwd}
+		}
+
+		// TODO: make sure we get the workspace configured.
+		if usrCfg.Workspace == "" {
+			cwd, err := os.Getwd()
+			BailOnError(err)
+			usrCfg.Workspace = filepath.Dir(filepath.Dir(cwd))
+		}
+
+		ws := workspace.New(usrCfg.Workspace)
+		tx, err := workspace.NewTransmission(ws.Dir, args)
+		BailOnError(err)
+
+		dirs, err := ws.Locate(tx.Dir)
+		BailOnError(err)
+
+		sx, err := workspace.NewSolutions(dirs)
+		BailOnError(err)
+
+		var solution *workspace.Solution
+		for {
```

Comment:
We only show the prompt if we actually found more than one. If we only found one we just pick it and return. It's not super obvious, but it let me make the code a little bit more straight forward, so I went with it.